pallets/xyk/src/lib.rs

```diff
@@ -11,92 +8,285 @@ mod mock;
+    use sp_runtime::traits::{Hash, StaticLookup};
+
+    use core::cmp::Ord;
 
     /// Configure the pallet by specifying the parameters and types on which it depends.
     #[pallet::config]
-    pub trait Config: frame_system::Config {
-        /// Because this pallet emits events, it depends on the runtime's definition of an event.
+    pub trait Config: frame_system::Config + pallet_assets::Config {
         type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        type AssetId: IsType<<Self as pallet_assets::Config>::AssetId> + Parameter + Ord + Copy;
+        /// Address where dex stores all the funds
+        type DEXAddr: Get<<Self::Lookup as StaticLookup>::Target>;
+        /// Minimum balance for issued LP tokens
+        type LpMinBalance: Get<Self::Balance>;
+        /// Hash for asset ids. It is used to determenistically get LP asset id from 2 asset ids.
+        type AssetIdHash: Hash;
     }
 
+    pub type AssetId<T> = <T as Config>::AssetId;
+    pub type Balance<T> = <T as pallet_assets::Config>::Balance;
+
     #[pallet::pallet]
     #[pallet::generate_store(pub(super) trait Store)]
     pub struct Pallet<T>(_);
 
-    // The pallet's runtime storage items.
-    // https://docs.substrate.io/v3/runtime/storage
+    #[derive(Debug, Clone, Copy, Encode, Decode, TypeInfo)]
+    pub struct Pair<Id, B> {
+        x: (Id, B),
+        y: (Id, B),
+    }
+
     #[pallet::storage]
-    #[pallet::getter(fn something)]
-    // Learn more about declaring storage items:
-    // https://docs.substrate.io/v3/runtime/storage#declaring-storage-items
-    pub type Something<T> = StorageValue<_, u32>;
+    pub type LPPairs<T> = StorageMap<_, Blake2_128Concat, AssetId<T>, Pair<AssetId<T>, Balance<T>>>;
 
-    // Pallets use events to inform users when important changes are made.
-    // https://docs.substrate.io/v3/runtime/events-and-errors
     #[pallet::event]
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
     pub enum Event<T: Config> {
-        /// Event documentation should end with an array that provides descriptive names for event
-        /// parameters. [something, who]
-        SomethingStored(u32, T::AccountId),
+        Swapped {
+            who: T::AccountId,
+            from: (AssetId<T>, Balance<T>),
+            to: (AssetId<T>, Balance<T>),
+        },
+        Locked {
+            who: T::AccountId,
+            x: (AssetId<T>, Balance<T>),
+            y: (AssetId<T>, Balance<T>),
+            lp: (AssetId<T>, Balance<T>),
+        },
+        Unlocked {
+            who: T::AccountId,
+            x: (AssetId<T>, Balance<T>),
+            y: (AssetId<T>, Balance<T>),
+            lp: (AssetId<T>, Balance<T>),
+        },
+        LPMinted {
+            who: T::AccountId,
+            x: (AssetId<T>, Balance<T>),
+            y: (AssetId<T>, Balance<T>),
+            lp: (AssetId<T>, Balance<T>),
+        },
     }
 
     // Errors inform users that something went wrong.
```

Comment:
Comments should start with `///` and be present on all enum variants. Then all of this will be rendered nicely in polkadot.js apps UI.